src/System.Data.SqlClient/tests/ManualTests/SQL/DataStreamTest/DataStreamTest.cs

```diff
@@ -34,6 +36,33 @@ public static void RunAllTestsForSingleServer_TCP()
             RunAllTestsForSingleServer(DataTestUtility.TcpConnStr);
         }
 
+        [CheckConnStrSetupFact]
+        public static void DataSetWriteXmlTest()
```

Comment:
I don't thing that this test belongs in SqlClient tests. This should be System.Data.Common or with the Xml tests. 
One approach I can think of, create this dataset by connecting to server and load this dataset. After having an in-memory copy of DataSet, you could persist it in XML and then load the XML in the test. 
Test for failures before and after the change.